File: web/src/app/model/Tile.ts
import {TileStyle} from '../utils/TileStyle';
import {TileRotation} from './TileRotation';
import {Pawn} from './Pawn';

export class Tile {

  constructor(tileId: number, xCoordinate: number, yCoordinate: number, topWall = false, bottomWall = false, rightWall = false, leftWall = false,
              imgSrc = TileStyle.EMPTY, pawnDTO: Pawn = null, rotation = TileRotation.Zero) {
    this.tileId = tileId;
    this.xCoordinate = xCoordinate;
    this.yCoordinate = yCoordinate;
    this.topWall = topWall;
    this.bottomWall = bottomWall;
    this.rightWall = rightWall;
    this.leftWall = leftWall;
    this.imgSrc = imgSrc;
    this.pawnDTO = pawnDTO;
    this.rotation = rotation;
  }

  public readonly tileId: number;
  public xCoordinate: number | null;
  public yCoordinate: number | null;
  public topWall: boolean;
  public bottomWall: boolean;
  public rightWall: boolean;
  public leftWall: boolean;
  public imgSrc: TileStyle;
  public rotation: TileRotation;
  public pawnDTO: Pawn;
}

File: api/src/main/java/com/hva/nl/ewa/models/Tile.java
package com.hva.nl.ewa.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

@Entity
@Table(name = "tile")
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Tile implements Model {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @NotNull
    private long tileId;

    @NotNull
    private TileRotation rotation;

    @JsonIgnore
    @OneToOne(targetEntity = Pawn.class, fetch = FetchType.LAZY)
    @JoinColumn(name = "pawn_id")
    private Pawn pawn;

    @NotNull
    private int tileDefinition;

    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "game_id")
    private Game game;

    private Integer xCoordinate;

    private Integer yCoordinate;

    public Tile() {
    }

    public Tile(Pawn pawn, TileDefinition tileDefinition) {
        this.rotation = TileRotation.Zero;
        this.pawn = pawn;
        this.tileDefinition = tileDefinition.getTileDefinitionId();
    }

    public Tile(Pawn pawn, TileDefinition tileDefinition, int initialYCoordinate, int initialXCoordinate) {
        this.rotation = TileRotation.Zero;
        this.pawn = pawn;
        this.tileDefinition = tileDefinition.getTileDefinitionId();
        this.xCoordinate = initialXCoordinate;
        this.yCoordinate = initialYCoordinate;
    }

    public TileRotation getRotation() {
        return rotation;
    }

    public void setRotation(TileRotation rotation) {
        this.rotation = rotation;
    }

    public Pawn getPawn() {
        return pawn;
    }

    public void setPawn(Pawn pawn) {
        this.pawn = pawn;
    }


    public TileDefinition getTileDefinition() {
        return TileDefinition.GetTile(tileDefinition);
    }

    public void setTileDefinition(TileDefinition tileDefinition) {
        this.tileDefinition = tileDefinition.getTileDefinitionId();
    }

    public Integer getyCoordinate() {
        return yCoordinate;
    }

    public void setyCoordinate(Integer yCoordinate) {
        this.yCoordinate = yCoordinate;
    }

    public Integer getxCoordinate() {
        return xCoordinate;
    }

    public void setxCoordinate(Integer xCoordinate) {
        this.xCoordinate = xCoordinate;
    }

    public long getTileId() {
        return tileId;
    }

    public void setTileId(long tileId) {
        this.tileId = tileId;
    }

    public Game getGame() {
        return game;
    }

    public void setGame(Game game) {
        this.game = game;
    }

    public boolean hasCard(Card card) {
        return card.IsOnTile(this);
    }

}
